File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;
class Solution {
public int solution(int[] elements) {
    int n = elements.length;
    int count = 0;
    Set<Integer> set = new HashSet<>();
    
    for (int i = 0; i < n; i++) {
        int sum = 0;
        for (int j = i; j < i + n; j++) {
            sum += elements[j % n];
            if (set.add(sum)) {
                count++;
            }
        }
    }
    
    return count;
}
}

File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    sum_set = set() 

    for start in range(len(elements)):
        sum_num = 0  
        for end in range(start,start + len(elements)):
            sum_num += elements[end % len(elements)]
            sum_set.add(sum_num)
    return len(sum_set)